health-services/libraries/health-services-common/src/main/java/org/egov/common/data/query/builder/GenericQueryBuilder.java

```diff
@@ -68,6 +69,20 @@ static List<String> getFieldsWithCondition(Object object, QueryFieldChecker chec
                             String fieldName = field.getName();
                             paramsMap.put(fieldName, field.get(object));
                             whereClauses.add(String.format("%s=:%s", fieldName, fieldName));
+                        } else if (field.getType().isAssignableFrom(ArrayList.class)
+                                && field.getGenericType() instanceof ParameterizedType
+                                && ((ParameterizedType) field.getGenericType()).getActualTypeArguments()[0].equals(String.class)) {
+                            ArrayList<String> arrayList = (ArrayList<String>) field.get(object);
+                            if (arrayList != null && !arrayList.isEmpty()) {
+                                String fieldName = field.getName();
+                                StringBuilder value = new StringBuilder();
+                                for (String arrayValue : arrayList) {
+                                    String paramName = fieldName + "_" + arrayList.indexOf(arrayValue);
+                                    value.append(":" + paramName + ",");
```

Comment:
can use the index to not add the comma for the last element in the list instead of removing the last comma separately afterwards.

Reply:
Updated